pkg/async/notifications/implementations/sendgrid_emailer.go

```diff
@@ -0,0 +1,67 @@
+package implementations
+
+import (
+	"context"
+	"os"
+
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	runtimeInterfaces "github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+)
+
+type SendgridEmailer struct {
+	client        *sendgrid.Client
+	systemMetrics emailMetrics
+}
+
+func getEmailAddresses(addresses []string) []*mail.Email {
+	sendgridAddresses := make([]*mail.Email, len(addresses))
+	for idx, email := range addresses {
+		sendgridAddresses[idx] = mail.NewEmail("", email)
+	}
+	return sendgridAddresses
+}
+
+func getSendgridEmail(adminEmail admin.EmailMessage) *mail.SGMailV3 {
+	m := mail.NewV3Mail()
+
+	from := mail.NewEmail("Flyte Notifications", adminEmail.SenderEmail)
```

Comment:
what is "Flyte Notifications" used for? should this be configurable?

Reply:
you know how when you send an email, gmail shows you something like "Alice <alice@umbrella.com>".  I think it's just the first part of it, the "name".  I don't think it's worth it to make it configurable, since at least for sendgrid, it doesn't actually matter (should probably add a comment though). The sender is determined by the API key.